cmd/krel/cmd/push.go

```diff
@@ -235,10 +241,65 @@ func runPushBuild(opts *pushBuildOptions) error {
 		return errors.Errorf("unable to identify specified bucket for artifacts: %s", releaseBucket)
 	}
 
-	// Check if bucket exists.
-	if _, err = bucket.Attrs(context.Background()); err != nil {
+	// Check if bucket exists and user has permissions
+	perms, err := bucket.IAM().TestPermissions(context.Background(), []string{"storage.objects.create"})
+	if err != nil {
 		return errors.Wrap(err, "Unable to find release artifact bucket")
 	}
+	if len(perms) != 1 {
+		return errors.Errorf("GCP user must have at least storage.objects.create permissions on bucket %s", releaseBucket)
+	}
+
+	if err = util.RemoveAndReplaceDir(release.GCSStagePath); err != nil {
+		return errors.Wrap(err, "Unable remove and replace GCS staging directory.")
+	}
+
+	// Copy release tarballs
+	if err = util.CopyDirContents(release.ReleaseTarsPath, release.GCSStagePath); err != nil {
+		return errors.Wrap(err, "Unable to copy source directory into destination")
+	}
+
+	// Prepare helpful GCP scripts
+	gciPath := release.GetGCIPath()
+	if err = util.CopyFile(filepath.Join(release.GCEPath, "configure-vm.sh"), filepath.Join(release.GCSStagePath, "extra/gce"), true); err != nil {
+		return err
+	}
+	if err = util.CopyFile(filepath.Join(gciPath, "node.yaml"), filepath.Join(release.GCSStagePath, "extra/gce"), false); err != nil {
+		return err
+	}
+	if err = util.CopyFile(filepath.Join(gciPath, "master.yaml"), filepath.Join(release.GCSStagePath, "extra/gce"), false); err != nil {
+		return err
+	}
+	if err = util.CopyFile(filepath.Join(gciPath, "configure.sh"), filepath.Join(release.GCSStagePath, "extra/gce"), false); err != nil {
+		return err
+	}
+	if err = util.CopyFile(filepath.Join(gciPath, "shutdown.sh"), filepath.Join(release.GCSStagePath, "extra/gce"), true); err != nil {
+		return err
+	}
+
+	// Prepare helpful Windows scripts
+	if err = util.CopyFile(filepath.Join(release.WindowsLocalPath, "configure.ps1"), release.WindowsGCSPath, false); err != nil {
+		return err
+	}
+	if err = util.CopyFile(filepath.Join(release.WindowsLocalPath, "common.psm1"), release.WindowsGCSPath, false); err != nil {
+		return err
+	}
+	if err = util.CopyFile(filepath.Join(release.WindowsLocalPath, "k8s-node-setup.psm1"), release.WindowsGCSPath, false); err != nil {
+		return err
+	}
+	if err = util.CopyFile(filepath.Join(release.WindowsLocalPath, "testonly/install-ssh.psm1"), release.WindowsGCSPath, false); err != nil {
+		return err
+	}
+	if err = util.CopyFile(filepath.Join(release.WindowsLocalPath, "testonly/user-profile.psm1"), release.WindowsGCSPath, false); err != nil {
+		return err
+	}
+
+	// TODO
+	// Prepare naked binaries
+	// Write checksums
+	// Push Docker images
```

Comment:
I think this should eventually just become a post-submit that builds and pushes containers in GCB instead of having to have it here.